### DISCUSSION TITLE
Using localization

### DISCUSSION THREAD

**Discussion by jegork** [fief-dev/fief#88] [Q&A] [answered]
Hey everyone!

I haven't been able to change localization for the login page. I can see that there is [i18next-browser-languageDetector](https://github.com/i18next/i18next-browser-languageDetector) used, but I haven't been able to change the localization using any of the methods provided there (neither query string nor local storage worked). Only changing the browser language led to localization change. Is there any way I can programmatically pass the localization value when redirecting to the login page?

Thanks!

**Comment by fief-bailiff** [comment]
Hail, @jegork 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment]
The authentication pages are generated by the Python server and are not handled by the React frontend, so i18next won't be used in this context.

Currently, the locale selection is handled by [asgi-babel](
https://github.com/klen/asgi-babel/blob/453a4681b3f2cd5f608dcbc13c328c0183cb0cb7/asgi_babel/__init__.py#L133-L168) which parses the `Accept` header from the HTTP request (i.e. the setting of the browser).

It's simple, but maybe it doesn't cover every use cases. Would you mind elaborating on how would you like to set the language for the login pages?

**Reply by jegork** [reply]
Well, either having a subpath or query string parameter would be sufficient I guess. But I guess it would be very handy to have this possibility of setting the language programmatically, as it would integrate better with most of the websites, where there is an ability to change the language using buttons.

**Reply by frankie567** [reply]
Right, I agree that it's an important feature for localization. I'll add it to the backlog!

**Comment by frankie567** [comment, marked as answer]
Hi @jegork 👋 

I just wished to inform you that with the latest version of Fief, you can add the `lang` parameter when generating the authorization URL so you can manually set the user locale on auth pages. Our official clients have been updated to handle this parameter:

* https://github.com/fief-dev/fief-python/releases/tag/v0.15.0
* https://github.com/fief-dev/fief-js/releases/tag/v0.11.0

**Reply by jegork** [reply]
Hey @frankie567 
That's great! Thank you for updating me!
